### ISSUE TITLE
export-name flags on files with a single export

### ISSUE THREAD

**Issue by astorije** [Microsoft/tslint-microsoft-contrib#448]
*Making a real issue out of https://github.com/Microsoft/tslint-microsoft-contrib/issues/121#issuecomment-384477314*

In `test/helpers/enzyme-intl.tsx`, I have:

```tsx
import { mount } from 'enzyme';
import React, { ReactElement } from 'react';
import { IntlProvider } from 'react-intl';

export function mountWithIntl<P>(component: ReactElement<P>) {
  return mount(<IntlProvider locale="en">{component}</IntlProvider>);
}
```

This triggers:

```
.../test/helpers/enzyme-intl.tsx:5:1
ERROR: 5:1  export-name  The exported module or identifier name must match the file name. Found: .../test/helpers/enzyme-intl.tsx and mountWithIntl
```

However, I tried to build a dummy test case to identify the violation and potentially post a question somewhere, but couldn't reproduce it on a different example:

- `test/helpers/foobar-helpers.ts`:
  ```ts
  export function foo(): string {
    return 'foo';
  }
  ```
- `test/helpers/foobar-helpers.tsx`:
  ```tsx
  import React, { ReactNode } from 'react';

  const Foo = () => <span>Foo</span>;

  export function foo(): ReactNode {
    return <Foo />;
  }
  ```

Both of these pass the `export-name` rule!
Is there something obvious I am missing?!

**Comment by JoshuaKGoldberg** [comment]
@astorije `export-name` looks like it's intended to allow exports whose names are the file name plus any extension, but this behavior implies it's allowing any general subset of the file name.

In your error case:
* `mountWithIntl` is _not_ a subset of `enzym-intl.tsx` ❌

In both your test cases:
* `foo` is a subset of `foobar-helpers.ts` ✔
* `foo` is a subset of `foobar-helpers.tsx`✔

Would you prefer strict matching or subset? (should this be a config option, maybe?)

For future reference:

https://github.com/Microsoft/tslint-microsoft-contrib/blob/73cfcdff42a7d1cc62b2de249e39863b441c7a98/src/exportNameRule.ts#L127

**Comment by astorije** [comment]
> Would you prefer strict matching or subset? (should this be a config option, maybe?)

I'm not sure, because at the moment I'm failing to see the value of this rule altogether. We are considering to disable it, but before doing so, would you mind clarifying its intent overall?

**Comment by JoshuaKGoldberg** [comment]
I believe the idea was to enforce consistent naming of exports and their files. Generally, most or all files should have a single purpose and therefore a single "thing" exported. If you adhere to this and have accurate & descriptive names for both your files and your objects/classes, then it follows that they should have similar names.

For example, if you have a file `Multiplier.ts`, it's natural to expect that it would have an export named `Multiplier`, but it would be weird having `Calculator` exported from it. Such a naming disconnect would be a sign that either your file is misnamed (and should be `Calculator.ts`) or your object should be renamed to `Calculator`.

❌ `import { Multiplier } from "./Calculator";`
✔ `import { Multiplier } from "./Multiplier";`
✔ `import { Calculator } from "./Calculator";

**Comment by JoshuaKGoldberg** [comment]
_(closing for housekeeping - thanks for discussing!)_
